lte/gateway/c/session_manager/SessionStateEnforcer.cpp

```diff
@@ -296,15 +297,17 @@ void SessionStateEnforcer::m5g_start_session_termination(
       *session, magma::lte::M5GSMCause::OPERATION_SUCCESS,
       PDU_SESSION_STATE_NOTIFY);
 
-  /* Call for all rules to be de-associated from session
-   * and inform to UPF
-   */
-  MLOG(MDEBUG) << "Will be removing all associated rules of session id "
+  if (previous_state != CREATING) {
+    /* Call for all rules to be de-associated from session
+     * and inform to UPF
+     */
+    MLOG(MDEBUG) << "Will be removing all associated rules of session id "
                << session->get_session_id();
-  m5g_pdr_rules_change_and_update_upf(session, PdrState::REMOVE);
-  if (session_map[imsi].size() == 0) {
-    // delete the rules
-    pdr_map_.erase(imsi);
+    m5g_pdr_rules_change_and_update_upf(session, PdrState::REMOVE);
+    if (session_map[imsi].size() == 0) {
+      // delete the rules
+      pdr_map_.erase(imsi);
+    }

```

Comment:
If rules are not there and we send a request, UPF will ignore it, isn't it?

But if we, for some reason, ever change the logic of the states or position of when rules are sent, and we forget to remove that condigion, then the rules may not be removed

Reply:
UPF is stateless process, means there is nothing to store in memory. Whatever SMF sends UPF should take action based on the incoming action. So in SMF we have to handle this un-necessary calling of removing rules if there are no rules in UPF.